extensions/binding/org.eclipse.smarthome.binding.lirc/src/main/java/org/eclipse/smarthome/binding/lirc/internal/connector/LIRCConnector.java

```diff
@@ -0,0 +1,211 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.lirc.internal.connector;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.PrintWriter;
+import java.net.Socket;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.io.IOUtils;
+import org.eclipse.smarthome.binding.lirc.internal.config.LIRCBridgeConfiguration;
+import org.eclipse.smarthome.binding.lirc.internal.messages.LIRCButtonEvent;
+import org.eclipse.smarthome.binding.lirc.internal.messages.LIRCResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Connector for communication with the LIRC server
+ *
+ * @author Andrew Nagle - Initial contributor
+ */
+public class LIRCConnector {
+
+    private static final Logger logger = LoggerFactory.getLogger(LIRCConnector.class);
+
+    private static List<LIRCEventListener> listeners = new ArrayList<LIRCEventListener>();
+
+    Socket socket;
+
+    InputStream in;
+    OutputStream out;
+
+    PrintWriter outWriter;
+
+    Thread readerThread;
+
+    public LIRCConnector() {
+
+    }
+
+    public synchronized void addEventListener(LIRCEventListener rfxComEventListener) {
+        if (!listeners.contains(rfxComEventListener)) {
+            listeners.add(rfxComEventListener);
+        }
+    }
+
+    public synchronized void removeEventListener(LIRCEventListener listener) {
+        listeners.remove(listener);
+    }
+
+    public void connect(LIRCBridgeConfiguration config) throws UnknownHostException, IOException {
+
+        logger.debug("Connecting");
+
+        // TODO: support unix sockets
```

Comment:
LIRC also supports [Unix domain sockets](https://en.wikipedia.org/wiki/Unix_domain_socket) in addition to TCP sockets. Unix domain sockets are the default way of communicating with LIRC, but that would require a third-party library and I wasn't really sure how to deal with licensing so I left this as a "phase two".